### PR TITLE
Feature header - Updates for Desktop design AEMCEDS-31

### PR DESCRIPTION
**Pull request by Abhishek-Tha**
Please always provide the [AEMCEDS-31] your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #<jira-issue-id>

Test URLs:
- Before: https://main--aem-competency--aem-comp.hlx.live/
- After: https://feature-header--aem-competency--aem-comp.hlx.live/

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://<branch>--aem-competency--aem-comp.hlx.live/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/<your-block>&index=0

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -179,7 +179,7 @@ a:any-link {
 }
 
 a:hover {
-  text-decoration: underline;
+  text-decoration: none;
```
Comment (mkbansal1) [inline styles/styles.css]: revert this rule, text decoration should underline for site, block specific experience should be overridden in block css.

File: blocks/header/header.js [hunk 2]
```diff
@@ -148,5 +185,17 @@ export default async function decorate(block) {
...
   navWrapper.append(nav);
+
+  const navDrops = navSections.querySelectorAll('.nav-drop');
+  navDrops.forEach((child) => {
+    const accordionArrow = document.createElement('i');
+    accordionArrow.className = 'arrow';
+    child.insertBefore(accordionArrow, child.firstChild);
+  });
+
+  if (window.innerWidth < 1280) {
```
Comment (mkbansal1) [inline blocks/header/header.js]: Akash has added a utility method in utils.js to check if viewport is desktop, use that

File: blocks/header/header.css [hunk 3]
```diff
@@ -177,10 +179,77 @@ header nav[aria-expanded="true"] .nav-sections {
   align-self: start;
 }
 
+/* Added for Mobile starts */
+@media (width < 1279px) {
```
Comment (mkbansal1) [inline blocks/header/header.css]: mobile view is <=1279

File: blocks/header/header.css [hunk 4]
```diff
@@ -177,10 +179,77 @@ header nav[aria-expanded="true"] .nav-sections {
...
+    header nav[aria-expanded="true"] .nav-sections {
+      background-color: black;
```
Comment (mkbansal1) [inline blocks/header/header.css]: Use variables instead of using colors directly

File: blocks/header/header.css [hunk 5]
```diff
@@ -115,6 +116,7 @@ header nav[aria-expanded="true"] .nav-hamburger-icon::after {
   width: 24px;
   height: 2px;
   border-radius: 2px;
+  z-index:100;
 }
```
Comment (mkbansal1) [inline blocks/header/header.css:108]: The background color should be transparent.
`background-color: transparent`

```
header nav .nav-hamburger button {
  height: 22px;
  margin: 0;
  border: 0;
  border-radius: 0;
  padding: 0;
  background-color: var(--background-color);
  color: inherit;
  overflow: initial;
  text-overflow: initial;
  white-space: initial;
}
```

**Review by mkbansal1 — CHANGES_REQUESTED** [review]
- Mobile view is totally broken
- Desktop view > dropdown hover effects are missing
- Desktop > down icon animation on hover is different from designs

File: blocks/header/header.js (hunk 2 above)
Comment (Abhishek-Tha) [reply blocks/header/header.js]: Updated

File: blocks/header/header.css (hunk 4 above)
Comment (Abhishek-Tha) [reply blocks/header/header.css]: Updated

File: blocks/header/header.css (hunk 5 above)
Comment (Abhishek-Tha) [reply blocks/header/header.css:108]: Updated

File: blocks/header/header.css (hunk 3 above)
Comment (Abhishek-Tha) [reply blocks/header/header.css]: Updated

**Comment by mrugesh-adobe** [comment]
Desktop and Mobile UI issues are been fixed.

Attaching the screenshot of the same.

desktop

mobile-view

**Review by mrugesh-adobe — APPROVED** [review]

**Comment by mrugesh-adobe** [comment]
> * Mobile view is totally broken
> * Desktop view > dropdown hover effects are missing
> * Desktop > down icon animation on hover is different from designs

Fixes are done.

**Comment by mrugesh-adobe** [comment]
image

**Review by mkbansal1 — APPROVED** [review]

**Comment by mrugesh-adobe** [comment]
image
image

**Pull request merged**